File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {


        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException("bad args");
        }
        int d = 0;

        for (int i = 0; i < firstStrand.Length; i++)
        {
            var c = firstStrand[i];
            var c2 = secondStrand[i];

            if (c != c2)
            {
                ++d;
            }
        }

        return d;
    }
}

File: haskell/hamming/src/Hamming.hs
module Hamming (distance) where

boolToInt :: Bool -> Int
boolToInt b = if b then 1 else 0

distance :: String -> String -> Maybe Int
distance a b =
    if length a /= length b
        then Nothing
    else
        Just $ sum $ map boolToInt $ zipWith (/=) a b
